### DISCUSSION TITLE
Abstract UI related components into a new namespace

### DISCUSSION THREAD

**Discussion by hishamco** [OrchardCMS/OrchardCore#12068] [Ideas]
I'm thinking about moving any related UI stuff into a new namesapce say `OrchardCore.UI`, this might include themes, zones ... etc

For example `IAdminThemeService`, `ISiteThemeService` could be move there, this way we have a better organization for UI related things that could be consumed in various modules

**Comment by ns8482e** [comment]
May be good idea, May be not
 not sure how would you keep each module independent?

**Reply by hishamco** [reply]
It SHOULD be good ;) If the UI components encapsulated and abstracted into its own package the module(s) need to reference them and use them as normal class libraries

**Reply by ns8482e** [reply]
IMHO - It's good only if it's not a module. - the negative is if it's not a module then you can't have views in shared library.

If it's a module then sure it's not a good idea

**Comment by hishamco** [comment]
> IMHO - It's good only if it's not a module. 

Not all, It's simple package such as `OC.Infrastructure` for UI

**Reply by ns8482e** [reply]
The issue is views may be you can implement as shape function but it may be tedious

**Reply by hishamco** [reply]
May be the issue if we want to target Liquid, coz in ASP.NET Views Component or TagHelpers can be a plus to create resusable UI components
